pkg/genericapiserver/server_run_options.go

```diff
@@ -55,28 +64,104 @@ type ServerRunOptions struct {
 	SecurePort                int
 	ServiceClusterIPRange     net.IPNet // TODO: make this a list
 	ServiceNodePortRange      utilnet.PortRange
-	TLSCertFile               string
-	TLSPrivateKeyFile         string
+	StorageVersions           string
+	// The default values for StorageVersions. StorageVersions overrides
+	// these; you can change this if you want to change the defaults (e.g.,
+	// for testing). This is not actually exposed as a flag.
+	DefaultStorageVersions string
+	TLSCertFile            string
+	TLSPrivateKeyFile      string
 }
 
 func NewServerRunOptions() *ServerRunOptions {
 	return &ServerRunOptions{
-		APIGroupPrefix:       "/apis",
-		APIPrefix:            "/api",
-		BindAddress:          net.ParseIP("0.0.0.0"),
-		CertDirectory:        "/var/run/kubernetes",
-		EnableLogsSupport:    true,
-		EnableProfiling:      true,
-		EnableWatchCache:     true,
-		InsecureBindAddress:  net.ParseIP("127.0.0.1"),
-		InsecurePort:         8080,
-		LongRunningRequestRE: defaultLongRunningRequestRE,
-		MasterCount:          1,
-		MaxRequestsInFlight:  400,
-		MinRequestTimeout:    1800,
-		RuntimeConfig:        make(config.ConfigurationMap),
-		SecurePort:           6443,
+		APIGroupPrefix:         "/apis",
+		APIPrefix:              "/api",
+		BindAddress:            net.ParseIP("0.0.0.0"),
+		CertDirectory:          "/var/run/kubernetes",
+		EnableLogsSupport:      true,
+		EnableProfiling:        true,
+		EnableWatchCache:       true,
+		InsecureBindAddress:    net.ParseIP("127.0.0.1"),
+		InsecurePort:           8080,
+		LongRunningRequestRE:   defaultLongRunningRequestRE,
+		MasterCount:            1,
+		MaxRequestsInFlight:    400,
+		MinRequestTimeout:      1800,
+		RuntimeConfig:          make(config.ConfigurationMap),
+		SecurePort:             6443,
+		StorageVersions:        registered.AllPreferredGroupVersions(),
+		DefaultStorageVersions: registered.AllPreferredGroupVersions(),
+	}
+}
+
+// StorageGroupsToEncodingVersion returns a map from group name to group version,
+// computed from the s.DeprecatedStorageVersion and s.StorageVersions flags.
+func (s *ServerRunOptions) StorageGroupsToEncodingVersion() (map[string]unversioned.GroupVersion, error) {
```

Comment:
We're only using this to build a `ResourceEncodingConfig`, right?  Should we refactor this to return that interface?